app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,32 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription ID is provided,
+      # then only collect data for that subscription. Otherwise, collect all data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
+
+        subscriptions.collect do |sub_id|
+          arm_service.configuration.subscription_id = sub_id
+
+          begin
+            if method.to_s == 'list_all'
+              arm_service.send(method).select { |obj| obj.location == @ems.provider_region }
```

Comment:
Adding or subtracting? Probably no clear winner depending on the real usage. Pulling large amount of unneeded data over the wire can be worse even though the number of requests have been reduced. It also depends on MS internal implementation - even though we only make one request, MS internally may request multiple times across regions.